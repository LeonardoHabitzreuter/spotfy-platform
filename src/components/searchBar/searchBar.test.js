import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from './index'

describe('SearchBar component', () => {
  test('should render properly', () => {
    const wrapper = shallow(<SearchBar />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should call onSearch when Icon was clicked', () => {
    const searchMock = jest.fn()
    const wrapper = shallow(<SearchBar onSearch={searchMock} />)
    const icon = wrapper.find('#icon')
    const input = wrapper.find('#input')

    input.simulate('change', { target: { value: 'param' } })
    icon.simulate('click')

    expect(searchMock).toHaveBeenCalledWith('param')
  })
})
