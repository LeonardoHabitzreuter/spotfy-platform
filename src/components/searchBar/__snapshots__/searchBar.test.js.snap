// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should render properly 1`] = `
<div
  className="input-group"
>
  <div
    className="input-group-prepend clickable"
  >
    <span
      className="input-group-text"
    >
      <Icon
        className="fas fa-search"
        id="icon"
        onClick={[Function]}
      />
    </span>
  </div>
  <input
    autoFocus={true}
    className="form-control searchInput"
    id="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
  />
</div>
`;
